src/data/real/ennreal.lean

```diff
@@ -1268,6 +1273,104 @@ begin
   exact n.lt_two_pow
 end
 
+@[simp, norm_cast] lemma coe_zpow (hr : r ≠ 0) (n : ℤ) : (↑(r^n) : ℝ≥0∞) = r^n :=
+begin
+  cases n,
+  { simp only [int.of_nat_eq_coe, coe_pow, zpow_coe_nat] },
+  { have : r ^ n.succ ≠ 0 := pow_ne_zero (n+1) hr,
+    simp only [zpow_neg_succ_of_nat, coe_inv this, coe_pow] }
+end
+
+lemma zpow_pos (ha : a ≠ 0) (h'a : a ≠ ∞) (n : ℤ) : 0 < a ^ n :=
+begin
+  cases n,
+  { exact ennreal.pow_pos ha.bot_lt n },
+  { simp only [h'a, pow_eq_top_iff, zpow_neg_succ_of_nat, ne.def, not_false_iff,
+               inv_pos, false_and] }
+end
+
+lemma zpow_lt_top (ha : a ≠ 0) (h'a : a ≠ ∞) (n : ℤ) : a ^ n < ∞ :=
+begin
+  cases n,
+  { exact ennreal.pow_lt_top h'a.lt_top _ },
+  { simp only [ennreal.pow_pos ha.bot_lt (n + 1), zpow_neg_succ_of_nat, inv_lt_top] }
+end
+
+lemma exists_zpow_near
+  {x y : ℝ≥0∞} (hx : x ≠ 0) (h'x : x ≠ ∞) (hy : 1 < y) (h'y : y ≠ ⊤) :
+  ∃ n : ℤ, y ^ n ≤ x ∧ x < y ^ (n + 1) :=
+begin
+  lift x to ℝ≥0 using h'x,
+  lift y to ℝ≥0 using h'y,
+  have A : y ≠ 0, by simpa only [ne.def, coe_eq_zero] using (ennreal.zero_lt_one.trans hy).ne',
+  obtain ⟨n, hn, h'n⟩ : ∃ n : ℤ, y ^ n ≤ x ∧ x < y ^ (n + 1),
+  { refine nnreal.exists_zpow_near _  (one_lt_coe_iff.1 hy),
+    simpa only [ne.def, coe_eq_zero] using hx },
+  refine ⟨n, _, _⟩,
+  { rwa [← ennreal.coe_zpow A, ennreal.coe_le_coe] },
+  { rwa [← ennreal.coe_zpow A, ennreal.coe_lt_coe] }
+end
+
+lemma exists_zpow_near'
+  {x y : ℝ≥0∞} (hx : x ≠ 0) (h'x : x ≠ ∞) (hy : 1 < y) (h'y : y ≠ ⊤) :
+  ∃ n : ℤ, y ^ n < x ∧ x ≤ y ^ (n + 1) :=
```

Comment:
Can you add a comment explaining how this is different to the unprimed version? Maybe it makes sense to state this as
```suggestion
lemma exists_zpow_near'
  {x y : ℝ≥0∞} (hx : x ≠ 0) (h'x : x ≠ ∞) (hy : 1 < y) (h'y : y ≠ ⊤) :
  ∃ n : ℤ, x ∈ Ioc (y ^ n) (y ^ (n + 1)) :=
```
so that you can use `Ioc` and `Ico` in the name